zng/string.go

```diff
@@ -35,17 +41,45 @@ func (t *TypeOfString) String() string {
 	return "string"
 }
 
-func (t *TypeOfString) StringOf(zv zcode.Bytes) string {
-	//XXX we need to rework this to conform with ZNG spec.
-	// for now, we are leaving binary data in the string here
-	// and leaving it up to the caller to escape as desired.
-	// (at least when we change this to bsring).
-	// XXX need to make sure we don't double escape in zio
-	//return EscapeUTF8(zv)
-	return string(zv)
+func uescape(r rune) []byte {
+	code := strconv.FormatInt(int64(r), 16)
+	var s string
+	if len(code) == 4 {
+		s = fmt.Sprintf("\\u%s", code)
+	} else {
+		s = fmt.Sprintf("\\u{%s}", code)
+	}
+	return []byte(s)
+}
+
+func (t *TypeOfString) StringOf(zv zcode.Bytes, ofmt OutFmt) string {
```

Comment:
mininit: name use `fmt` instead of `ofmt` for consistency

Reply:
(or ofmt in the other `StringOf`s

Reply:
Ah, in an earlier draft I was using the golang fmt module and the names collided but that's no longer the case, will fix.